app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/CommonTriageQuestionsController.scala

```diff
@@ -419,15 +420,86 @@ class CommonTriageQuestionsController @Inject() (
               else routes.SingleDisposalsTriageController.whenWasCompletionDate()
           }
 
-        Ok(
-          previousReturnExistsWithSameCompletionDatePage(
-            state.fold(_.subscribedDetails, _.subscribedDetails).isATrust,
-            backLink
-          )
-        )
+        withCompletionDate(state) { value =>
+          state.fold(
+            _.previousSentReturns
+              .map(_.summaries)
+              .getOrElse(List.empty)
+              .find(e => CompletionDate(e.completionDate) === value),
+            _.previousSentReturns
+              .map(_.summaries)
+              .getOrElse(List.empty)
+              .find(e => CompletionDate(e.completionDate) === value)
+          ) match {
+            case Some(matchingPreviousReturn) if state.fold(_ => true, r => !r.isAmendReturn) =>
+              Ok(
+                previousReturnExistsWithSameCompletionDatePage(
+                  matchingPreviousReturn,
+                  backLink
+                )
+              )
+            case _ if state.fold(_ => false, _.isAmendReturn)                                 =>
+              Ok(
+                previousReturnExistsWithSameCompletionDateAmendPage(
+                  value,
+                  state.fold(_.subscribedDetails, _.subscribedDetails).isATrust,
+                  backLink
+                )
+              )
+            case _                                                                            => Redirect(uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.routes.StartController.start())
+          }
+        }
       }
     }
 
+  private def withCompletionDate(state: Either[StartingNewDraftReturn, FillingOutReturn])(
+    f: CompletionDate => Future[Result]
+  ): Future[Result] =
+    state match {
+      case Left(value)  =>
```

Comment:
having `value` everywhere here is a bit confusing - it'd be better to have more descriptive names